lib/Reader/readerir.cpp

```diff
@@ -7499,3 +7662,381 @@ void VerificationState::print() {
 #endif
 
 #pragma endregion
+
+#pragma region SIMD_INTRISNICS
+
+//===----------------------------------------------------------------------===//
+//
+// SIMD Intrinsics
+//
+//===----------------------------------------------------------------------===//
+
+Type *GenIR::FloatTy;
+Type *GenIR::FloatPtrTy;
+Type *GenIR::Vector2Ty;
+Type *GenIR::Vector3Ty;
+Type *GenIR::Vector4Ty;
+Type *GenIR::Vector2PtrTy;
+Type *GenIR::Vector3PtrTy;
+Type *GenIR::Vector4PtrTy;
+
+// BinOperations
+
+IRNode *GenIR::vectorAdd(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFAdd(Vector1, Vector2);
+}
+
+IRNode *GenIR::vectorSub(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFSub(Vector1, Vector2);
+}
+
+IRNode *GenIR::vectorMul(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFMul(Vector1, Vector2);
+}
+
+IRNode *GenIR::vectorDiv(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFDiv(Vector1, Vector2);
+}
+
+IRNode *GenIR::vectorEqual(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFCmpOEQ(Vector1, Vector2);
+}
+
+IRNode *GenIR::vectorNotEqual(IRNode *Vector1, IRNode *Vector2) {
+  return (IRNode *)LLVMBuilder->CreateFCmpONE(Vector1, Vector2);
```

Comment:
Are the `F{Add,Sub,Mul,Div,CmpOEQ,CmpONE}` instructions used above always correct, or do we need to worry about vectors with integer elements? Does that only come up with generic vectors?